src/pngwriter.cc

```diff
@@ -1633,35 +1633,35 @@ int pngwriter::read_png_info(FILE *fp, png_structp *png_ptr, png_infop *info_ptr
 
 ////////////////////////////////////////////////////////////
 int pngwriter::read_png_image(FILE *fp, png_structp png_ptr, png_infop info_ptr,
-			      png_bytepp *image, png_uint_32 *width, png_uint_32 *height)
+                              png_bytepp *image, png_uint_32& width, png_uint_32& height)
 {
    unsigned int i,j;
 
-   *width = png_get_image_width(png_ptr, info_ptr);
-   *height = png_get_image_height(png_ptr, info_ptr);
+   width = png_get_image_width(png_ptr, info_ptr);
+   height = png_get_image_height(png_ptr, info_ptr);
 
-   if( width == NULL)
+   if( width == 0 )
```

Comment:
an alternative solution that would probably also satisfy the Coverity Scan CID is using `*width == 0` here.